agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -575,8 +585,18 @@ bool monitor_thread::update_sta_stats()
             std::chrono::duration_cast<std::chrono::milliseconds>(now - sta_stats.last_update_time);
         sta_stats.delta_ms         = float(time_span.count());
         sta_stats.last_update_time = now;
+
+        if (std::chrono::steady_clock::now() > awake_timeout()) {
+            // If we haven't finish to iterate on all stations, skip one time on the select timeout
+            // so the select will not be stuck on full select timeout and the thread will be able to
+            // finish this operation quickly.
+            skip_next_select_timeout();
+            return true;
+        }
```

Comment:
I would move this condition above the call to `update_stations_stats`.
Otherwise it could trigger the `skip_next_select_timeout` even if we just processed the last station in this cycle.

Reply:
moved.